### PR TITLE
added a parity test between 20180930preview and internal API

### LINKED ISSUE #933: add unit tests for internal and external representation parity with appropriate exceptions
There should be an explicit parity unit test similar to TestAdminAPIParity in pkg/api, but for the External representation.
This will allow for a more concise error messages in unit tests and separates it from converter unit tests.

### PR DESCRIPTION
**Pull request by Makdaam**
Fixes #933 
Not sure if the AuthProfile.IdentityProviders Secret field should be missing from the internal representation. The test ignores that difference for now.

### CODE REVIEW

File: pkg/api/types_test.go [hunk 1]
```diff
@@ -416,3 +417,82 @@ loop:
 		t.Fatalf("mismatch between internal and admin API fields: afield=%q, ifield=%q", afield, ifield)
 	}
 }
+
+func Test20180930previewAPIParity(t *testing.T) {
+	// the algorithm is: list the fields of both types.  If the head of either
```
Comment (kwoodson) [inline pkg/api/types_test.go]: Good comment!

Comment (Makdaam) [reply pkg/api/types_test.go]: All credit to @jim-minter who wrote the Admin API parity test which I copied and adapted.

File: pkg/api/types_test.go [hunk 2]
```diff
@@ -416,3 +417,82 @@ loop:
...
+				reflect.TypeOf(AADIdentityProvider{}),
+			},
+		},
+	)
+
+	// TODO: I don't believe this should be in the admin type at all
+	notInInternal := []*regexp.Regexp{
+		regexp.MustCompile(`^\.Response`),
+		regexp.MustCompile(`^\.Properties\.MasterPoolProfile`),
+		regexp.MustCompile(`^\.Properties\.AuthProfile\.IdentityProviders\.Provider\.Secret`),
```
Comment (kargakis) [inline pkg/api/types_test.go]: This field should exist in both the internal and the customer api

File: pkg/api/types_test.go [hunk 3]
```diff
@@ -416,3 +417,82 @@ loop:
...
+	}
+
+	// TODO: why don't we just include all of these in the admin type?
+	notInV20180930preview := []*regexp.Regexp{
+		regexp.MustCompile(`^\.Properties\.ServicePrincipalProfile`),
```
Comment (kargakis) [inline pkg/api/types_test.go]: This field should exist in both the internal and the customer api

File: pkg/api/types_test.go [hunk 4]
```diff
@@ -416,3 +417,82 @@ loop:
...
+		regexp.MustCompile(`^\.Properties\.AzProfile`),
```
Comment (kargakis) [inline pkg/api/types_test.go]: ditto

Comment (kargakis) [reply pkg/api/types_test.go]: Sorry, servicePrincipalProfile and azProfile are not included in the customer profile, disregard

File: pkg/api/types_test.go (hunk 2 above)
Comment (Makdaam) [reply pkg/api/types_test.go]: Thanks for catching this, there was a second list of exceptions, I removed that and everything matches my expectations now.

**Comment by kargakis** [comment]
https://github.com/openshift/openshift-azure/issues/949

/retest

**Pull request merged**